metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -145,11 +147,11 @@ final class RenameProvider(
       }
       val fileChanges = allChanges.flatten.toMap
       val shouldRenameInBackground =
-        !metalsConfig.openFilesOnRenames ||
+        (!metalsConfig.openFilesOnRenames && !clientExperimentalCapabilities.openFilesOnRenameProvider) ||
           fileChanges.keySet.size >= metalsConfig.renameFileThreshold
       val (openedEdits, closedEdits) =
         if (shouldRenameInBackground) {
-          if (metalsConfig.openFilesOnRenames) {
+          if (metalsConfig.openFilesOnRenames || clientExperimentalCapabilities.openFilesOnRenameProvider) {
```

Comment:
Could we move it to a value? We can use that in both places then.

Reply:
Sure! Fixed.